apps/frontend/src/utils/math.ts

```diff
@@ -129,12 +134,13 @@ export const getLocaleSeparators = () => {
   };
 };
 
-export const getDecimalPartLength = (value: string | number) =>
-  value?.toString().split('.')?.[1]?.length || 0;
+export const getDecimalPartLength = (value: Decimalish) =>
+  Decimal.from(value).toString().split('.')?.[1]?.length || 0;
 
-export const numeric = (value: number) => {
-  if (isNaN(value) || !isFinite(value) || !value) {
-    return 0;
+export const numeric = (value: Decimalish | undefined | null) => {
```

Comment:
Make sure you use `numeric` instead of `Decimal.from` everywhere, it will look uniform + you already handle `undefined | null` in `numeric` so it's more error-proof.